MovieApp/MainActivity.cs

```diff
@@ -0,0 +1,516 @@
+                string retorno = CallRequest(endereco);
+
+                var movies = JsonConvert.DeserializeObject<Movie>(retorno);
+                foreach (var m in movies.results)
+                {
+
+                    try
+                    {
+                        contador++;
+
+                        string genero = "";
+                        for (int g = 0; g < m.genre_ids.Count; g++)
+                        {
+                            if (genero.Length > 0)
+                            {
+                                genero += "/";
+                            }
+                            var itemGenero = listaGenero.Where(x => x.idGenero == m.genre_ids[g]).ToList()[0];
+                            genero += itemGenero.genero;
+                        }
+
+                        //List of Details                
+                        string endereoDetail = "https://api.themoviedb.org/3/movie/" + m.id + "?api_key=" + apiKey + "&language=en-US";
+                        string responseFromServerImage = CallRequest(endereoDetail);
+                        var detail = JsonConvert.DeserializeObject<RootObjectDetail>(responseFromServerImage);
+                        //string caminhoImagem = detail.poster_path;
+                        string caminhoImagem = "?";
+
+                        items.Add(detail.title + " - " + caminhoImagem + " - " + genero + " - " + detail.release_date);
+                        itemsIndex.Add(detail.id + "-" + detail.title + " - " + caminhoImagem + " - " + genero + " - " + detail.release_date);
+
+                        Filme reg = new Filme();
+                        reg.Descricao = detail.title + " - " + caminhoImagem + " - " + genero + " - " + detail.release_date;
+                        reg.Id = detail.id;
+
+
+                        string im = "https://image.tmdb.org/t/p/w500" + detail.poster_path;
+
+
+                        Bitmap imageBitmap = null;
+                        using (var webClient = new WebClient())
+                        {
+                            var imageBytes = webClient.DownloadData(im);
+                            if (imageBytes != null && imageBytes.Length > 0)
+                            {
+                                imageBitmap = BitmapFactory.DecodeByteArray(imageBytes, 0, imageBytes.Length);
+                            }
+                        }
+                        reg.Poster = imageBitmap;
+
+                        FilmesRepositorio.Filmes.Add(reg);
+
+                    }
+                    catch (Exception ex)
+                    {
+
+                    }
+                }
+
+            }
+
+            var listAdapter = new ArrayAdapter<string>(this, Android.Resource.Layout.SimpleListItem1, items.ToArray());
+            list.Adapter = listAdapter;
+
+        }
+        private static string CallRequest(string endereco)

```

Comment:
Not making this an asynchronous call is really a terrible idea. On top of that there is a thread sleep that blocks the main thread even more.